emeis/core/tests/test_api.py

```diff
@@ -0,0 +1,216 @@
+            for query in query_context.captured_queries
+        ],
+        "request": {
+            k: v for k, v in response.request.items() if not k.startswith("wsgi")
+        },
+    }
+
+    # Drop `SAVEPOINT` statements because they will change on every run.
+    value["queries"] = list(
+        filter(lambda lst: "SAVEPOINT" not in lst, value["queries"])
+    )
+
+    if include_json:
+        value["response"] = response.json()
+
+    snapshot.assert_match(value)
+
+
+@pytest.mark.freeze_time("2017-05-21")
+def test_api_list(fixture, request, admin_client, snapshot, viewset):
+    url = reverse("{0}-list".format(viewset.base_name))
+
+    # create more data for proper num queries check
+    request.getfixturevalue(viewset.factory_name + "_factory").create_batch(2)
+
+    included = getattr(viewset.serializer_class, "included_serializers", {})
+    with CaptureQueriesContext(connection) as context:
+        response = admin_client.get(url, data={"include": ",".join(included.keys())})
+
+    assert_response(response, context, snapshot)
+
+
+@pytest.mark.freeze_time("2017-05-21")
+def test_api_detail(fixture, admin_client, viewset, snapshot):
+    url = reverse("{0}-detail".format(viewset.base_name), args=[fixture.pk])
+
+    included = getattr(viewset.serializer_class, "included_serializers", {})
+    with CaptureQueriesContext(connection) as context:
+        response = admin_client.get(url, data={"include": ",".join(included.keys())})
+
+    assert_response(response, context, snapshot)
+
+
+@pytest.mark.freeze_time("2017-05-21")
+def test_api_create(fixture, admin_client, viewset, snapshot):
+    url = reverse("{0}-list".format(viewset.base_name))
+
+    serializer = viewset.serializer_class(fixture)
+    renderer = JSONRenderer()
+    context = {"view": viewset}
+    data = renderer.render(serializer.data, renderer_context=context)
+    fixture.delete()  # avoid constraint issues
+
+    with CaptureQueriesContext(connection) as context:
+        response = admin_client.post(url, data=json.loads(data))
+
+    assert_response(response, context, snapshot)
+
+
+@pytest.mark.freeze_time("2017-05-21")
+def test_api_patch(fixture, admin_client, viewset, snapshot):
+    url = reverse("{0}-detail".format(viewset.base_name), args=[fixture.pk])
+
+    serializer = viewset.serializer_class(fixture)
+    renderer = JSONRenderer()
+    context = {"view": viewset}
+    data = renderer.render(serializer.data, renderer_context=context)
+
+    with CaptureQueriesContext(connection) as context:
+        response = admin_client.patch(url, data=json.loads(data))
+
+    assert_response(response, context, snapshot)
+
+
+@pytest.mark.freeze_time("2017-05-21")
+def test_api_destroy(fixture, admin_client, snapshot, viewset):
+    url = reverse("{0}-detail".format(viewset.base_name), args=[fixture.pk])
+
+    # may not assert queries till following issue is fixed:
+    # https://code.djangoproject.com/ticket/29559
```

Comment:
This is fixed upstream, so I think we could re-enable the call below. Might also do it for our customer project where this code comes from :)

Reply:
Done.